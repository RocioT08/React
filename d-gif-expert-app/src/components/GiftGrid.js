import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
import PropTypes from "prop-types";

const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Loading ...</p>
      )}
      {
        <div className="card-grid">
          {images.map((image) => (
            <GifGridItem key={image.id} {...image} />
          ))}
        </div>
      }
    </>
  );
};

GiftGrid.prototype = {
  category: PropTypes.string.isRequired,
};

export default GiftGrid;
