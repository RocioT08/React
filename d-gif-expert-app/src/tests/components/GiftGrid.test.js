import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GiftGrid from "../../components/GiftGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Test GiftGridItem Component", () => {
  const category = "goku";

  test("should render the component correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when useFetchGifs hook return data", () => {
    const images = [
      {
        id: "abc",
        title: "image_title",
        url: "http://localhost/img/test.jpg",
      },
      {
        id: "abcd",
        title: "image_title_2",
        url: "http://localhost/img/test_2.jpg",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: images,
      loading: false,
    });
    const wrapper = shallow(<GiftGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(images.length);
  });
});
