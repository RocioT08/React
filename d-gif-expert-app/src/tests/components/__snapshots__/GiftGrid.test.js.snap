// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GiftGridItem Component should render the component correctly 1`] = `
<Fragment>
  <h3>
    goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test GiftGridItem Component should show items when useFetchGifs hook return data 1`] = `
<Fragment>
  <h3>
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="image_title"
      url="http://localhost/img/test.jpg"
    />
    <GifGridItem
      id="abcd"
      key="abcd"
      title="image_title_2"
      url="http://localhost/img/test_2.jpg"
    />
  </div>
</Fragment>
`;
